opensoundscape/annotations.py

```diff
@@ -178,87 +186,88 @@ def from_raven_files(
+                    df = df.reindex(columns=standard_columns)
+                else:
+                    # keep all columns
+                    pass
 
-            # rename Raven columns to standard opensoundscape names
-            try:
-                df = df.rename(
-                    columns=column_mapping_dict,
-                    errors="raise",
-                )
-            except KeyError as e:
-                raise KeyError(
-                    "Raven file is missing a required column. "
-                    "Raven files must have columns matching the following names: "
-                    f"{column_mapping_dict.keys()}"
-                ) from e
-
-            # add column containing the raven file path
-            df["annotation_file"] = raven_file
-
-            # remove undesired columns
-            standard_columns = [
-                "annotation_file",
-                "start_time",
-                "end_time",
-                "low_f",
-                "high_f",
-            ]
-            if annotation_column_idx is not None:
-                standard_columns.append("annotation")
-            if hasattr(keep_extra_columns, "__iter__"):
-                # keep the desired columns
-                # if values in keep_extra_columns are missing, fill with nan
-                df = df.reindex(
-                    columns=standard_columns + list(keep_extra_columns),
-                    fill_value=np.nan,
-                )
-            elif not keep_extra_columns:
-                # only keep required columns
-                df = df.reindex(columns=standard_columns)
-            else:
-                # keep all columns
-                pass
+                # add audio file column
+                if audio_files is not None:
+                    df["audio_file"] = audio_files[i]
+                else:
+                    df["audio_file"] = np.nan
 
-            # add audio file column
-            if audio_files is not None:
-                df["audio_file"] = audio_files[i]
+                all_file_dfs.append(df)
             else:
-                df["audio_file"] = np.nan
+                warnings.warn(f"{raven_file} has zero rows.")
 
-            all_file_dfs.append(df)
+        if len(all_file_dfs) > 0:
+            # we drop the original index from the Raven annotations when we combine tables
+            # if the dataframes have different columns, we fill missing columns with nan values
+            # and keep all unique columns
+            all_annotations_df = pd.concat(all_file_dfs).reset_index(drop=True)
 
-        # we drop the original index from the Raven annotations when we combine tables
-        # if the dataframes have different columns, we fill missing columns with nan values
-        # and keep all unique columns
-        all_annotations = pd.concat(all_file_dfs).reset_index(drop=True)
+        else:
+            all_annotations_df = pd.DataFrame(
```

Comment:
I had chosen to hardcode these columns because the `required_cols` variable was only defined within the scope of the `__init__` function.  Probably the same reason why you defined `standard_columns` twice. I changed it  [here](https://github.com/kitzeslab/opensoundscape/commit/6e3f60980accd383b4fddc40b13becdfdc500cdc), but did not change the `standard_columns` variable because I was afraid of breaking something.